File: spec/python/test_instance_io_user.py
import unittest

from instance_io_user import InstanceIoUser

class TestInstanceIoUser(unittest.TestCase):
    def test_instance_io_user(self):
        r = InstanceIoUser.from_file("src/instance_io.bin")

        self.assertEquals(r.qty_entries, 3)

        self.assertEquals(r.entries[0].name, "the")
        self.assertEquals(r.entries[1].name, "rainy")
        self.assertEquals(r.entries[2].name, "day it is")

File: spec/lua/test_instance_io_user.lua
local luaunit = require("luaunit")

require("instance_io_user")

TestInstanceIoUser = {}

function TestInstanceIoUser:test_instance_io_user()
    local r = InstanceIoUser:from_file("src/instance_io.bin")

    luaunit.assertEquals(r.qty_entries, 3)

    luaunit.assertEquals(r.entries[1].name, "the")
    luaunit.assertEquals(r.entries[2].name, "rainy")
    luaunit.assertEquals(r.entries[3].name, "day it is")
end
